wiki/Spotlight Guide.md

```diff
@@ -0,0 +1,142 @@
+
+## Hooking up the spotlight
+In order to request a spotlight in the UI, use [SpotlightManager](https://github.com/oppia/oppia-android/blob/d2c37dc547f3e5d12dfe62fa97b9b16fbf0fed6e/app/src/main/java/org/oppia/android/app/spotlight/SpotlightManager.kt#L4):
+
+```kotlin
+checkNotNull(getSpotlightManager()).requestSpotlight(
+	SpotlightTarget(
+	binding.yourFeatureName,
+	R.string.your_feature_string_string_resource,
+	SpotlightShape.RoundedRectangle,
+	Spotlight.FeatureCase.YOUR_FEATURE_NAME
+)
+)
+
+  private fun getSpotlightManager(): SpotlightManager? {
+    return fragment.requireActivity().supportFragmentManager.findFragmentByTag(
+      SpotlightManager.SPOTLIGHT_FRAGMENT_TAG
+    ) as? SpotlightManager
+  }
+```
+
+For views that are laid out late in the lifecycle (such as recycler views loaded after a data provider call, view pagers, or elements that show up after an enter animation), use [requestSpotlightWithDelayedLayout](https://github.com/oppia/oppia-android/blob/d2c37dc547f3e5d12dfe62fa97b9b16fbf0fed6e/app/src/main/java/org/oppia/android/app/spotlight/SpotlightManager.kt#L13) instead.
+
+# Testing spotlights
+Spotlights are tested by checking if the expected hint is shown on the screen or not.
+
+```kotlin
+  @Test
+  fun testPromotedStorySpotlight_setToShowOnSecondLogin_notSeenBefore_checkSpotlightShown() {
+    logIntoUserTwice()
+    launch<HomeActivity>(createHomeActivityIntent(internalProfileId1)).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(withText(R.string.promoted_story_spotlight_hint))
+        .check(matches(isDisplayed()))
+      // Or, use check(doesNotExist()) to verify it is not being shown.
+    }
+  }
+```
+
+In order to bypass a spotlight without any UI interaction, the ``SpotlightStateController`` should be used to mark the spotlights as seen. A helper function can be introduced and used to mark one or more spotlights as seen. 
+	
+```kotlin
+  private fun markAllSpotlightsSeen() {
+    val profileId = ProfileId.newBuilder().setInternalId(internalProfileId).build()
+    spotlightStateController.markSpotlightViewed(profileId, TOPIC_LESSON_TAB)  
+    spotlightStateController.markSpotlightViewed(profileId, TOPIC_REVISION_TAB) 
+    spotlightStateController.markSpotlightViewed(profileId, YOUR_FEATURE_NAME)
+    testCoroutineDispatchers.runCurrent()
+  }
+```
+
+## Disabling the spotlights
+If the spotlights need to be disabled, the platform parameter value should be set.
+If the spotlights need to be disabled in the tests, use the ``TestPlatformParameterModule`` to turn off the [enableSpotlightUi](https://github.com/oppia/oppia-android/blob/d2c37dc547f3e5d12dfe62fa97b9b16fbf0fed6e/utility/src/main/java/org/oppia/android/util/platformparameter/PlatformParameterConstants.kt#L218) platform parameter.
```

Comment:
Should this be a new paragraph? If so, add a newline above this one. Please check and do a self-review using the Markdown preview functionality on GitHub.

Reply:
done, PTAL